rosetta-lang/src/main/java/com/regnosys/rosetta/formatting2/XtextResourceFormatter.java

```diff
@@ -0,0 +1,77 @@
+package com.regnosys.rosetta.formatting2;
+
+import java.nio.charset.StandardCharsets;
+import java.util.Collection;
+import java.util.List;
+import java.io.ByteArrayInputStream;
+import java.io.IOException;
+import java.io.InputStream;
+import java.io.UncheckedIOException;
+
+import org.eclipse.emf.ecore.resource.Resource;
+import org.eclipse.xtext.formatting2.FormatterRequest;
+import org.eclipse.xtext.formatting2.IFormatter2;
+import org.eclipse.xtext.formatting2.regionaccess.ITextRegionAccess;
+import org.eclipse.xtext.formatting2.regionaccess.ITextRegionRewriter;
+import org.eclipse.xtext.formatting2.regionaccess.ITextReplacement;
+import org.eclipse.xtext.formatting2.regionaccess.TextRegionAccessBuilder;
+import org.eclipse.xtext.preferences.ITypedPreferenceValues;
+import org.eclipse.xtext.resource.XtextResource;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import javax.inject.Inject;
+import javax.inject.Provider;
+
+public class XtextResourceFormatter implements ResourceFormatterService{
+	private static Logger LOGGER = LoggerFactory.getLogger(XtextResourceFormatter.class);
+	@Inject
+	private Provider<FormatterRequest> formatterRequestProvider;
+	
+	@Inject 
+	private Provider<IFormatter2> iFormatter2Provider;
+
+	@Inject
+	private TextRegionAccessBuilder regionBuilder;
+
+	@Override
+	public void formatCollection(Collection<Resource> resources, ITypedPreferenceValues preferenceValues){
+		resources.stream().forEach(resource -> {
+			if (resource instanceof XtextResource) {
+	            formatXtextResource((XtextResource) resource, preferenceValues);
+	            
+	        } else {
+	        	LOGGER.debug("Resource is not of type XtextResource and will be skipped: " + resource.getURI());
+	        }
+		});
+	}
+
+	@Override
+	public void formatXtextResource(XtextResource resource, ITypedPreferenceValues preferenceValues){
+		//setup request and formatter
+		FormatterRequest req = formatterRequestProvider.get();
+		req.setPreferences(preferenceValues);
+		IFormatter2 formatter = iFormatter2Provider.get();
+		
+		ITextRegionAccess regionAccess = regionBuilder.forNodeModel(resource).create();
+		req.setTextRegionAccess(regionAccess);
+		
+		//list contains all the replacements which should be applied to resource
+		List<ITextReplacement> replacements = formatter.format(req);
+				
+		//formatting using TextRegionRewriter
+		ITextRegionRewriter regionRewriter = regionAccess.getRewriter();
+		String formattedString = regionRewriter.renderToString(regionAccess.regionForDocument(), replacements);
+
+		//With the formatted text, update the resource
+		InputStream resultStream = new ByteArrayInputStream(formattedString.getBytes(StandardCharsets.UTF_8));
+		resource.unload();
+		try {
+			resource.load(resultStream, null);
+			resource.save(null);
```

Comment:
This is interesting. I think it would be good to keep the formatting in-memory only, and then users of the class can decide whether they wanna actually save it or not. That way it might cover more use cases.

So what I would do is move this `save` into your main class, and perhaps also document that the formatting is done in-memory.